src/Microsoft.AspNet.Mvc.Core/ProducesAttribute.cs

```diff
@@ -74,12 +75,12 @@ public override void OnResultExecuting(ResultExecutingContext context)
             }
         }
 
-        private List<MediaTypeHeaderValue> GetContentTypes(string firstArg, string[] args)
+        private MediaTypeCollection GetContentTypes(string firstArg, string[] args)
         {
             var completeArgs = new List<string>();
             completeArgs.Add(firstArg);
             completeArgs.AddRange(args);
-            var contentTypes = new List<MediaTypeHeaderValue>();
+            var contentTypes = new MediaTypeCollection();
             foreach (var arg in completeArgs)
             {
                 var contentType = MediaTypeHeaderValue.Parse(arg);
```

Comment:
This is actually 
var contentType = new StringSegment(arg);
if (MediaTypeComparisons.MatchesAllSubtypes(contentType)||
    MediaTypeComparisons.MatchesAllTypes(contentType))